dll/win32/shell32/iconcache.cpp

```diff
@@ -905,6 +920,19 @@ UINT WINAPI ExtractIconExA(LPCSTR lpszFile, INT nIconIndex, HICON * phiconLarge,
     {
         MultiByteToWideChar(CP_ACP, 0, lpszFile, -1, lpwstrFile, len);
         ret = ExtractIconExW(lpwstrFile, nIconIndex, phiconLarge, phiconSmall, nIcons);
+
+        /*
+         * ExtractIconExW() which is based upon PrivateExtractIconEx and the internal routine, ICO_ExtractIconEx, return -1
+         * if the file is not valid (PE file image not being valid) or the file has not been found. As the -W variant (Unicode)
+         * of the function, it only returns the successful extracted icons from a file and in such case, return 0 for no successful
+         * extracted icons.
+        */
+        if (ret == 0xFFFFFFFF)
```

Comment:
ExtractIconExW won't return this value, since that's what your change above fixes. So this check is unnecessary.